pkg/api/stewardship_test.go

```diff
@@ -92,19 +96,28 @@ func Test_stewardshipHandlers_invalidInputs(t *testing.T) {
 				},
 			},
 		},
+	}, {
+		name:    "batch - does not exist",
+		address: "1234",
+		want: jsonhttp.StatusResponse{
+			Code:    http.StatusNotFound,
+			Message: "batch with id not found",
+		},
 	}}
 
 	for _, method := range []string{http.MethodGet, http.MethodPut} {
 		method := method
-		for _, tc := range tests {
+		for idx, tc := range tests {
 			tc := tc
-			t.Run(method+" "+tc.name, func(t *testing.T) {
-				t.Parallel()
+			if idx < 2 || method == http.MethodPut {
```

Comment:
what is the reason for having this `if` block on line 112?

Reply:
The test no. 3 is only applicable to MethodPut.

Reply:
would it be possible to add additional field to test cases structs with headers that each test case supports?

Reply:
I have moved it to a separate test. Dont think there is any advantage of making these changes.